h2o-core/src/main/java/water/fvec/Vec.java

```diff
@@ -471,6 +469,83 @@ public Vec makeCopy(String[] domain, byte type) {
     return v;
   }
 
+  /**
+   * Remaps vec's current domain levels to a new set of values. The cardinality of the new set of domain values might be
+   * less than or equal to the cardinality of current domain values. The new domain set is automatically extracted from
+   * the given mapping array.
+   * <p>
+   * Changes are made to this very vector, no copying is done. If you need the original vector to remain unmodified,
+   * please make sure to copy it first.
+   *
+   * @param newDomainValues An array of new domain values. For each old domain value, there must be a new value in
+   *                        this array. The value at each index of newDomainValues array represents the new mapping for
+   *                        this very index. May not be null.
+   * @throws UnsupportedOperationException When invoked on non-categorical vector
+   * @throws IllegalArgumentException      Length of newDomainValues must be equal to length of current domain values of
+   *                                       this vector
+   */
+  public void remapDomain(final String[] newDomainValues) throws UnsupportedOperationException, IllegalArgumentException {
```

Comment:
IMHO this is a too high-level function to be included in Vec, I would move it out of this class. 

Take a look at function `setDomain` - this function just changes the domain on the live object but doesn't reflect the changes in DKV.

This function mutates the Vec's data doesn't make sure the Vec's metadata is also updated. That is why I think it doesn't belong to the Vec class.

Reply:
At the end of the function
```
this.setMeta(Vec.T_CAT, reducedDomain);
```

Reply:
Absolutely right. Addressed.